### PR TITLE
Add description field to option block object

### PR DESCRIPTION
**Pull request by deto-r**
In this PR, I added a description field in the option object.
With this change, description can be used in select box etc. 

Other minor corrections below
・Corrected because there was a typo
・Added checkboxes test function
・Remove unnecessary type conversions in test files

### CODE REVIEW

File: block_element.go [hunk 1]
```diff
@@ -387,7 +387,7 @@ func (c CheckboxGroupsBlockElement) ElementType() MessageElementType {
 	return c.Type
 }
 
-// NewCheckboxGroupsBlockElement returns an instance of a radio block element
+// NewCheckboxGroupsBlockElement returns an instance of a checkbox-group block element
```
Comment (kanata2) [inline block_element.go:390]: 👍

File: block_element_test.go [hunk 2]
```diff
@@ -128,11 +152,11 @@ func TestNewRadioButtonsBlockElement(t *testing.T) {
...
 	// Build each option, providing a value for the option
-	radioButtonsOptionOne := NewOptionBlockObject("value-0", radioButtonsOptionTextOne)
-	radioButtonsOptionTwo := NewOptionBlockObject("value-1", radioButtonsOptionTextTwo)
-	radioButtonsOptionThree := NewOptionBlockObject("value-2", radioButtonsOptionTextThree)
+	radioButtonsOptionOne := NewOptionBlockObject("value-0", radioButtonsOptionTextOne, nil)
+	radioButtonsOptionTwo := NewOptionBlockObject("value-1", radioButtonsOptionTextTwo, nil)
+	radioButtonsOptionThree := NewOptionBlockObject("value-2", radioButtonsOptionTextThree, nil)
 
-	// Build overflow section
+	// Build radio button element
```
Comment (kanata2) [inline block_element_test.go:159]: thanks!

**Review by kanata2 — APPROVED** [review]
Sorry for slow response. Thank you!

**Comment by kanata2** [comment]
@deepoil I overlokked API breaking changes. So I reverted it, and will submit new PR which fixes a typo `mrkdown`.

**Comment by kanata2** [comment]
@deepoil cc: @ollieparsley
I'm sorry. I overlooked API breaking changes and revert it.
Could you submit new PR again for more detailed review?

**Comment by deto-r** [comment]
@kanata2 
I got an error in examples/modal_users/users.go, so I fixed it additionally. 
Other than that, I made another PR([#783 ](https://github.com/slack-go/slack/pull/783)) with the same content.

**Comment by kanata2** [comment]
Thank you!

**Pull request merged**